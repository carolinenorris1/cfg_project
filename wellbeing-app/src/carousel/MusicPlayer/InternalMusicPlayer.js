import { useEffect, useState } from "react";
import InternalPlayer from "./InternalPlayer";

function InternalMusicPlayer() {
  const [songs] = useState([
    {
      title: "A Promise",
      artist: "Keys of Moon",
      img_src: "./images/music-player-images/song-1.jpeg",
      src: "./music/a-promise.mp3",
    },
    {
      title: "Cat's Cradle",
      artist: "Purrple Cat",
      img_src: "./images/music-player-images/song-4.jpeg",
      src: "./music/Cats-Cradle.mp3",
    },
    {
      title: "Memories of Spring",
      artist: "Tokyo Music Walker",
      img_src: "./images/music-player-images/song-3.jpeg",
      src: "./music/Memories-of-Spring.mp3",
    },
    {
      title: "Midnight Stroll",
      artist: "Ghostrifter Official",
      img_src: "./images/music-player-images/song-5.jpeg",
      src: "./music/Midnight-Stroll-Lofi-Study-Music.mp3",
    },
    {
      title: "Otjanbird Pt II",
      artist: "SpheriÃ¡",
      img_src: "./images/music-player-images/song-6.jpeg",
      src: "./music/Otjanbird-Pt.-II.mp3",
    },
    {
      title: "Sunset Landscape",
      artist: "Keys of Moon",
      img_src: "./images/music-player-images/song-2.jpeg",
      src: "./music/Sunset-Landscape.mp3",
    },
  ]);

  const [currentSongIndex, setCurrentSongIndex] = useState(0);
  const [nextSongIndex, setNextSongIndex] = useState(currentSongIndex + 1);

  useEffect(() => {
    setNextSongIndex(() => {
      if (currentSongIndex + 1 > songs.length - 1) {
        return 0;
      } else {
        return currentSongIndex + 1;
      }
    });
  }, [currentSongIndex]);

  return (
    <div className="App">
      <InternalPlayer
        currentSongIndex={currentSongIndex}
        setCurrentSongIndex={setCurrentSongIndex}
        nextSongIndex={nextSongIndex}
        songs={songs}
      />
    </div>
  );
}

export default InternalMusicPlayer;
