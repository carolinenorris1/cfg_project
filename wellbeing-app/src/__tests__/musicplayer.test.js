import React from "react";
import renderer from "react-test-renderer";

import InternalMusicPlayer from "../carousel/MusicPlayer/InternalMusicPlayer";

it("renders correctly when user views the music player on the carousel", () => {
  const tree = renderer.create(<InternalMusicPlayer />).toJSON();
  expect(tree).toMatchSnapshot();
});
