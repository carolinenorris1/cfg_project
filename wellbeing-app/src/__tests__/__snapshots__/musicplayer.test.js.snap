// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when user views the music player on the carousel 1`] = `
<div
  className="App"
>
  <div
    className="c-player"
  >
    <audio
      src="./music/a-promise.mp3"
    />
    <div
      className="player-center"
    >
       
      <div
        className="c-player--details"
      >
        <div
          className="details-img"
        >
          <img
            alt=""
            src="./images/song-1.jpeg"
          />
        </div>
        <h3
          className="details-title"
        >
          A Promise
        </h3>
        <h4
          className="details-artist"
        >
          Keys of Moon
        </h4>
      </div>
      <div
        className="c-player--controls"
      >
        <button
          className="skip-btn"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-backward "
            data-icon="backward"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.5 440.6c9.5 7.9 22.8 9.7 34.1 4.4s18.4-16.6 18.4-29V96c0-12.4-7.2-23.7-18.4-29s-24.5-3.6-34.1 4.4L288 214.3V256v41.7L459.5 440.6zM256 352V256 128 96c0-12.4-7.2-23.7-18.4-29s-24.5-3.6-34.1 4.4l-192 160C4.2 237.5 0 246.5 0 256s4.2 18.5 11.5 24.6l192 160c9.5 7.9 22.8 9.7 34.1 4.4s18.4-16.6 18.4-29V352z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
        <button
          className="play-btn"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play "
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
        <button
          className="skip-btn"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-forward "
            data-icon="forward"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
      <p>
        Next up: 
        <span>
          Cat's Cradle
           by 
          Purrple Cat
        </span>
      </p>
    </div>
  </div>
</div>
`;
